projects/gateway2/proxy_syncer/endpoints_prioritize.go

```diff
@@ -48,14 +53,29 @@ func NewPriorities(failoverPriority []string) *Prioritizer {
 	}
 }
 
-func (p *Prioritizer) GetPriority(proxyLabels, upstreamEndpointLabels map[string]string) int {
+func (p *Prioritizer) GetPriority(proxyLabels, upstreamEndpointLabels map[string]string, defaultNetwork string) int {
```

Comment:
This function, with all its conditional logic, seems like it would benefit from some comprehensive unit tests to prove the differnt logical conditions. wdyt?

Reply:
I think the unit test I added covers every case

Reply:
can confirm the tests are useful, when I rewrote the impl the tests failed due to various mistakes (inverted ==/!=, etc)